app/code/Magento/QuoteGraphQl/Model/Resolver/Cart/CreateEmptyCart.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\QuoteGraphQl\Model\Resolver\Cart;
+
+use Magento\Framework\GraphQl\Config\Element\Field;
+use Magento\Framework\GraphQl\Query\Resolver\Value;
+use Magento\Framework\GraphQl\Query\Resolver\ValueFactory;
+use Magento\Framework\GraphQl\Query\ResolverInterface;
+use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
+use Magento\Quote\Api\CartManagementInterface;
+use Magento\Quote\Api\GuestCartManagementInterface;
+use Magento\Quote\Model\QuoteIdMask;
+use Magento\Quote\Model\QuoteIdMaskFactory;
+
+/**
+ * {@inheritdoc}
+ */
+class CreateEmptyCart implements ResolverInterface
+{
+    /**
+     * @var CartManagementInterface
+     */
+    private $cartManagement;
+
+    /**
+     * @var GuestCartManagementInterface
+     */
+    private $guestCartManagement;
+
+    /**
+     * @var ValueFactory
+     */
+    private $valueFactory;
+
+    /**
+     * @var QuoteIdMaskFactory
+     */
+    private $quoteIdMaskFactory;
+
+    /**
+     * @param CartManagementInterface $cartManagement
+     * @param GuestCartManagementInterface $guestCartManagement
+     * @param ValueFactory $valueFactory
+     * @param QuoteIdMaskFactory $quoteIdMaskFactory
+     */
+    public function __construct(
+        CartManagementInterface $cartManagement,
+        GuestCartManagementInterface $guestCartManagement,
+        ValueFactory $valueFactory,
+        QuoteIdMaskFactory $quoteIdMaskFactory
+    ) {
+        $this->cartManagement = $cartManagement;
+        $this->guestCartManagement = $guestCartManagement;
+        $this->valueFactory = $valueFactory;
+        $this->quoteIdMaskFactory = $quoteIdMaskFactory;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function resolve(Field $field, $context, ResolveInfo $info, array $value = null, array $args = null) : Value
+    {
+        $customerId = $context->getUserId();
```

Comment:
We should avoid working with global `$context` object, it is possible to inject `UserContextInterface` via constructor instead.